client/src/pages/WeeklyTable.js

```diff
@@ -2,21 +2,26 @@ import React from "react";
 
 import TableHeading from "./TableHeading";
 
-const WeeklyTable = (props) => {
-	return (
-		<div>
-			<table className="table">
-				<TableHeading />
-				<tbody>
-					{props.result.map((info, index) => (
-						<tr key={index}>
-							<td>{info.username}</td>
-						</tr>
-					))}
-				</tbody>
-			</table>
-		</div>
-	);
-}
+const fillInRow = (seat, bookings) =>
+	bookings
+		.filter((booking) => booking.desk_user_id === seat)
+		.map((b, index) => (
+			<td key={index}>
+				{b.username}
+			</td>
+		));
+
+const WeeklyTable = (props) => (
+	<div>
+		<table>
+			<TableHeading />
+			<tbody>
+				<tr>{fillInRow(1, props.result)}</tr>
```

Comment:
Taking the data, we have, in order of bookings

- // monday
- Maha
- // tuesday 
- // nobody
- // wednesday
- Sam
- Maha
- Pablo
- // thursday
- Pablo
- Maha
- // friday
- Maha

On the first row, we have Maha, ___, Sam, Pablo, Maha
on the second row we have ___, ___, Maha, Maha, ___
on the third row we have ___, ___, Pablo, ___, ___

What does the first row have in common?
What does the 2nd row have in common?
What does the 3rd row have in common?

How would the following data structures help? What code could you write to create them from the bookings array?

bookingsByDay = {
  "2022-01-17T00:00:00.000Z": [{id: 1, name:"Maha", date: "2022-01-17T00:00:00.000Z"}],
  "2022-01-18T00:00:00.000Z": [],
// I would preserve the whole original booking in the array, 
// so that you have the id available
// but I just wrote strings below to make this more readable in a comment
  "2022-01-19T00:00:00.000Z": ["Sam", "Maha", "Pablo"],
  "2022-01-20T00:00:00.000Z": ["Pablo","Maha"],
  "2022-01-21T00:00:00.000Z": ["Maha"],
}

// same here, keep the whole booking including booking id
bookingsByRow = [
["Maha",undefined,"Sam","Pablo","Maha"],
[undefined,undefined,"Maha","Maha",undefined],
[undefined,undefined,"Pablo",undefined,undefined],
]

Reply:
Hint of how I would implement this here: do on your own before looking at my code (also I haven't tried to execute it so it may be completely wrong)!!!!! https://gist.github.com/gregdyke/a723217826fdc37e239dd81f64d9a88b